clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/controllers/ManifestController.java

```diff
@@ -124,31 +114,24 @@ Manifest getDynamicManifestFromCluster(
         String.format(
             "(account: %s, location: %s, kind: %s, app %s, cluster: %s, criteria: %s)",
             account, location, kind, app, cluster, criteria);
-    List<List<? extends Manifest>> manifestSet =
-        manifestProviders.stream()
-            .map(
-                p -> {
-                  try {
-                    return requestQueue.execute(
-                        account,
-                        () ->
-                            p.getClusterAndSortAscending(
-                                account, location, kind, app, cluster, criteria.getSort()));
-                  } catch (Throwable t) {
-                    log.warn("Failed to read {}", request, t);
-                    return null;
-                  }
-                })
-            .filter(l -> l != null && !l.isEmpty())
-            .collect(Collectors.toList());
-
-    if (manifestSet.isEmpty()) {
+
+    List<KubernetesV2Manifest> manifests;
+    try {
+      manifests =
+          requestQueue.execute(
+              account,
+              () ->
+                  manifestProvider.getClusterAndSortAscending(
+                      account, location, kind, app, cluster, criteria.getSort()));
+    } catch (Throwable t) {
+      log.warn("Failed to read {}", request, t);
+      return null;
+    }
+
+    if (manifests.isEmpty()) {
```

Comment:
Should this still be `manifests == null` here?  I think that before we were only throwing an exception if all of the manifest providers returned `null` (indicating that none of them could even handle the request), but we didn't throw if at least one returned an empty list (indicating it handled the request but didn't find clusters).  I think this will make us throw an exception if there are no clusters.  But I also might be mis-understanding as the prior logic of `List<List<? extends Manifest>>` as kind of confusing.

Reply:
No, I was confused, great catch...added one more commit marking `getClusterAndSortAscending` as nullable & changing this to a null check. We're handling the case of it being empty or containing too few elements below where we're wrapping the criteria check in a try/catch.